### PR TITLE
Document the naming convention for `block-library` PHP functions

### LINKED ISSUE #53732: Naming convention for PHP functions in the `block-library` package should be documented
## What problem does this address?
https://github.com/WordPress/gutenberg/issues/52769#issuecomment-1669540435 needs to be documented in the `block-library` documentation.
https://github.com/WordPress/gutenberg/pull/53438 enforces the rule outlined in https://github.com/WordPress/gutenberg/issues/52769#issuecomment-1669540435, but there is no documentation for that naming convention yet.

Props: @gziolo

## What is your proposed solution?
The appropriate `readme.md` file has to be updated.

### PR DESCRIPTION
**Pull request by anton-vlasenko**
## What?

This PR is intended to document the following naming convention: https://github.com/WordPress/gutenberg/issues/52769#issuecomment-1669540435.
Fixes https://github.com/WordPress/gutenberg/issues/53732.

## Why?

Documenting every naming convention is essential.

## How?

- [x] Add the naming convention to the documentation.
- [x] Include a link to the documentation in `ValidBlockLibraryFunctionNameSniff`.

## Testing Instructions
Please review the suggested changes for accuracy, grammar, and style errors.

### CODE REVIEW

File: packages/block-library/README.md [hunk 1]
```diff
@@ -119,4 +119,23 @@ To find out more about contributing to this package or Gutenberg as a whole, ple
     }
     ```
 
+### Naming convention for PHP functions
+
+All PHP function names within the subdirectories of the `packages/block-library/src/` directory should start with one of the following prefixes:
```
Comment (gziolo) [inline packages/block-library/README.md]: Should it be clarified that it applies to function definitions? Example:
```suggestion
All declared PHP function names within the subdirectories of the `packages/block-library/src/` directory should start with one of the following prefixes:
```

**Review by gziolo — APPROVED** [review]
Thank you for the follow-up. I had only a minor note. Otherwise, this looks great.

File: packages/block-library/README.md (hunk 1 above)
Comment (anton-vlasenko) [reply packages/block-library/README.md]: Good catch. Fixed in 0586c6351a3cc255078ee5aea63915b644a87c42.

**Comment by anton-vlasenko** [comment]
> Thank you for the follow-up. I had only a minor note. Otherwise, this looks great.

Thank you for the code review, @gziolo. I appreciate it.

**Pull request merged**